tools/src/main/java/org/apache/kafka/tools/consumer/TransactionLogMessageFormatter.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.tools.consumer;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.MessageFormatter;
+import org.apache.kafka.common.protocol.ByteBufferAccessor;
+import org.apache.kafka.coordinator.transaction.generated.TransactionLogKey;
+import org.apache.kafka.coordinator.transaction.generated.TransactionLogKeyJsonConverter;
+import org.apache.kafka.coordinator.transaction.generated.TransactionLogValue;
+import org.apache.kafka.coordinator.transaction.generated.TransactionLogValueJsonConverter;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.ByteBuffer;
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class TransactionLogMessageFormatter implements MessageFormatter {
+
+    private static final String VERSION = "version";
+    private static final String DATA = "data";
+
+
+    @Override
+    public void writeTo(ConsumerRecord<byte[], byte[]> consumerRecord, PrintStream output) {
+        ObjectNode json = new ObjectNode(JsonNodeFactory.instance);
+        String content;
+        
+        byte[] key = consumerRecord.key();
+        if (Objects.nonNull(key)) {
+            short keyVersion = ByteBuffer.wrap(key).getShort();
+            Optional<TransactionLogKey> transactionLogKey = readToTransactionLogKey(ByteBuffer.wrap(key));
+            settingKeyNode(json, transactionLogKey, keyVersion);
+        } else {
+            return;
```

Comment:
That should be impossible that null key has associated value in this case, but we can keep going to parse the value, right? If we want to return early, it would be great to have comments for the scenario.

Reply:
@chia7712, Thanks for your comments, There is a problem, Is it possible that only have transaction log  value, but don't have the transaction log key scenario? I think we should give an error message but don't print the value node

Reply:
> I think we should give an error message but don't print the value node

this tool is used to print the contents of internal topic. Personally, I'd like to print what it sees since that is useful to debug. If we add something check (i.e null key should not have value), it would be hard to debug it if something trouble happens. WDYT?

Reply:
It is important to show enough message to debug, I think that null key show message should as same as null value

Reply:
> It is important to show enough message to debug

yep and that is why I prefer to show the value even though the key is null